mapstructure_test.go

```diff
@@ -556,6 +556,38 @@ func TestDecode_EmbeddedArray(t *testing.T) {
 	}
 }
 
+func TestDecode_decodeSliceWithArray(t *testing.T) {
```

Comment:
I'm sorry but I don't know how to add tests in this repository.
If this isn't good, please let me know.
I'll fix.